tests/cli-inst-interactive.rs

```diff
@@ -266,3 +268,112 @@ fn test_warn_if_complete_profile_is_used() {
+        assert!(out
+            .stderr
+            .contains("warning: it looks like you have existing rustup.sh metadata"));
+        assert!(out
+            .stderr
+            .contains("error: cannot install while rustup.sh is installed"));
+        assert!(out.stdout.contains("Continue? (y/N)"));
+    })
+}
+
+#[test]
+fn test_prompt_fail_if_rustup_sh_already_installed_reply_no() {
+    setup(&|config| {
+        create_rustup_sh_metadata(&config);
+        let out = run_input(config, &["rustup-init"], "no\n");
+        assert!(!out.ok);
+        assert!(out
+            .stderr
+            .contains("warning: it looks like you have existing rustup.sh metadata"));
+        assert!(out
+            .stderr
+            .contains("error: cannot install while rustup.sh is installed"));
+        assert!(out.stdout.contains("Continue? (y/N)"));
+    })
+}
+
+#[test]
+fn test_prompt_succeed_if_rustup_sh_already_installed_reply_yes() {
+    setup(&|config| {
+        create_rustup_sh_metadata(&config);
+        let out = run_input(config, &["rustup-init"], "yes\n\n\n");
+        assert!(out.ok);
+        assert!(out
+            .stderr
+            .contains("warning: it looks like you have existing rustup.sh metadata"));
+        assert!(out
+            .stderr
+            .contains("error: cannot install while rustup.sh is installed"));
+        assert!(out.stdout.contains("Continue? (y/N)"));
+        assert!(!out.stdout.contains(
+            "warning: continuing (because the -y flag is set and the error is ignorable)"
+        ))
+    })
+}
+
+#[test]
+fn test_warn_succeed_if_rustup_sh_already_installed_y_flag() {
+    setup(&|config| {
+        create_rustup_sh_metadata(&config);
+        let out = run_input(config, &["rustup-init", "-y"], "");
+        assert!(out.ok);
+        assert!(out
+            .stderr
+            .contains("warning: it looks like you have existing rustup.sh metadata"));
+        assert!(out
+            .stderr
+            .contains("error: cannot install while rustup.sh is installed"));
+        assert!(out.stderr.contains(
+            "warning: continuing (because the -y flag is set and the error is ignorable)"
+        ));
+        assert!(!out.stdout.contains("Continue? (y/N)"));
+    })
+}
+
+#[test]
+fn test_succeed_if_rustup_sh_already_installed_env_var_set() {
+    setup(&|config| {
+        create_rustup_sh_metadata(&config);
+        let mut cmd = clitools::cmd(config, "rustup-init", &["-y"]);
+        clitools::env(config, &mut cmd);
+        cmd.env("RUSTUP_INIT_SKIP_EXISTENCE_CHECKS", "yes");
+
+        cmd.stdin(Stdio::piped());
+        cmd.stdout(Stdio::piped());
+        cmd.stderr(Stdio::piped());
+        let _out = cmd.spawn().unwrap().wait_with_output().unwrap();
+        let out = SanitizedOutput {
+            ok: _out.status.success(),
+            stdout: String::from_utf8(_out.stdout).unwrap(),
+            stderr: String::from_utf8(_out.stderr).unwrap(),
+        };
```

Comment:
I feel like if this wasn't already possible via clitools more directly, then we need to improve that, rather than having this code in a test case.  It makes it harder to see what the test case is up to.

Reply:
This was copied from `run_input`, not `clitools`.

Reply:
I see.  It'd be preferable to make `run_input()` more generic (perhaps `run_input_with_env()` with `run_input()` changed to call through that) just so that we don't end up with code duplication for this stuff.

Reply:
Good idea.

Reply:
In your rebase I think you lost this bit of the test change.